File: my-app/src/lib/validations/gear.ts
import { z } from 'zod';

export const gearStatusEnum = z.enum(['owned', 'wishlist', 'sold', 'rented']);

export const gearCategoryEnum = z.enum([
  'camera',
  'lens',
  'lighting',
  'audio',
  'bag',
  'misc',
]);

export const createGearSchema = z.object({
  name: z.string().min(1, 'Name is required').max(200),
  category: gearCategoryEnum.nullable().optional(),
  brand: z.string().max(100).nullable().optional(),
  model: z.string().max(100).nullable().optional(),
  serial_number: z.string().max(100).nullable().optional(),
  purchase_price: z.number().nonnegative().nullable().optional(),
  purchase_date: z.string().date().nullable().optional(),
  status: gearStatusEnum.default('owned'),
  notes: z.string().nullable().optional(),
});

export const updateGearSchema = createGearSchema.partial().extend({
  id: z.string().uuid(),
});

export const listGearSchema = z.object({
  status: gearStatusEnum.optional(),
  category: gearCategoryEnum.optional(),
});

export type CreateGearInput = z.input<typeof createGearSchema>;
export type UpdateGearInput = z.input<typeof updateGearSchema>;
export type ListGearInput = z.input<typeof listGearSchema>;

File: ios/FocalsKit/Sources/FocalsModels/Gear.swift
import Foundation

public struct Gear: Codable, Identifiable, Hashable, Sendable {
    public let id: UUID
    public let userId: UUID
    public let name: String
    public let category: String?
    public let brand: String?
    public let model: String?
    public let serialNumber: String?
    public let purchasePrice: Double?
    public let purchaseDate: String?
    public let status: GearStatus?
    public let notes: String?
    public let createdAt: Date

    public init(
        id: UUID,
        userId: UUID,
        name: String,
        category: String?,
        brand: String?,
        model: String?,
        serialNumber: String?,
        purchasePrice: Double?,
        purchaseDate: String?,
        status: GearStatus?,
        notes: String?,
        createdAt: Date
    ) {
        self.id = id
        self.userId = userId
        self.name = name
        self.category = category
        self.brand = brand
        self.model = model
        self.serialNumber = serialNumber
        self.purchasePrice = purchasePrice
        self.purchaseDate = purchaseDate
        self.status = status
        self.notes = notes
        self.createdAt = createdAt
    }

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case name, category, brand, model
        case serialNumber = "serial_number"
        case purchasePrice = "purchase_price"
        case purchaseDate = "purchase_date"
        case status, notes
        case createdAt = "created_at"
    }
}
